### DISCUSSION TITLE
Connect your home node to BTCPay Server via LNC

### DISCUSSION THREAD

**Discussion by Liongrass** [btcpayserver/btcpayserver#4859] [Ideas / Feature Requests]
I'm a Lightning enthusiast (disclaimer: I also work for Lightning Labs). I run multiple node for various use cases, consuming considerable resources and capital in the process.
I also help others run Lightning nodes, including through BTCPay Server. Most commonly, the individuals and business owners I talk to, despite running their own nodes at home, dismiss BTCPay Server because of its significant resource overhead, including paying for a VPS with enough storage for a Bitcoin node, enough CPU power for a Lightning node and the time to manage liquidity. People already run their own nodes at home on their own hardware, which I think they should do! But how can they make use of BTCPay Server?

### The Remedy
Implement Lightning Node Connect (LNC) as a convenient option for people to connect their existing home node to their BTCPay Server installation on a VPS, even if their home node is behind tor. You can learn about how [LNC works under the hood here](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect), There are [npm](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lnc-npm) and [react native](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lnc-rn) packages that help with the integration. As BTCPay Server is almost guaranteed to run on clearnet behind a domain, it may also include [aperture](https://github.com/lightninglabs/aperture) to mitigate privacy issues that come from using an external proxy.

### Opening the path to a leaner BTCPay Server experience
Does every BTCPay Server need a Bitcoin and Lightning node inside of it? This is where the vast majority of resource overhead comes from when running your own instance. We could make it easier for people to run a BTCPay Server 'Lite', meaning without the Bitcoin or Lightning node inside. It would be ready to use almost instantly upon installation and connect to their existing node at home over LNC.

#### Share your BTCPay Server
Furthermore, such a setup would make it trivial to share your BTCPay Server installation with anyone who has a Lightning node at home. I currently use the LNbits plugin for that, but given my experiences a self-custodial solution would be superior in many ways.

#### Enhanced security with LND Accounts
LND Accounts is kind of like the LNbits plugin, but enforced by Macaroons directly by your node. Once I can connect to my own node via LNC, without any additional work, I will also be able to create a unique Account for BTCPay Server, for example limiting it to invoice-only, or segregating funds between my personal spending wallet and the funds BTCPay Server has access to. I can also create a new sub account for each of my stores, allowing for clearer accounting.

#### On-chain payments through my Lightning Node
On-chain payments might continue to be a popular request for a while, and I believe eventually they could be optionally handled through the Lightning node too. LND for instance already supports importing multiple xpubs and keep track of them in its internal wallet.

As a whole, I believe such a setup would make BTCPay Server easier to maintain, reduce cost and downtime. I would no longer need to manage liquidity for multiple nodes, juggle tailscale or SSH forwarding, I could let friends or community members use my BTCPay instance either with their home node, or a dedicated custodial account on my own node.

### Feature request
Despite the many words above, my feature request is simple. I humbly ask you to consider _LNC_ as a connection method for Lightning nodes. LNC is already available to all Umbrel, Raspiblitz, Voltage or Start9 users.

**Comment by beeforbacon** [comment]
Yes! This is needed ASAP.

**Comment by pavlenex** [comment]
This makes sense to me. @Liongrass Could you elaborate if any mobile wallets currently use LNC?

**Reply by beeforbacon** [reply]
Zeus and Alby does. And I just want to say BTCPayServer enabling LNC will go a long way to reducing friction of connecting an LND node to BTCPayServer. No more will we have to define hex macaroons, certs, etc., and thanks to the magic under the hood, specifically the mailbox proxy, tor nodes have access to clearnet. This is such a big level up and really excited for it to come to BTCPay as soon as possible.

**Comment by Liongrass** [comment]
[Zeus](https://docs.zeusln.app/for-users/remote-connections/umbrel#using-lightning-node-connect) uses Lightning Node Connect, and so does [Alby](https://blog.getalby.com/lightning-node-connect-in-the-alby-browser-extension/). I use both that way regularly. Connecting from my mobile phone to my node (behind Tor) is very smooth and doesn't require me to set up my own proxy.

**Comment by dstrukt** [comment]
Excited to see this will be getting more consideration!

**Comment by pavlenex** [comment]
Okay, I'm converting this one to the issue since there seems to be enough interest in implemetning this. I think it can be a good part of our efforts towards allowing more wallets to connect easier to btcpay, so the UX should follow accordingly. @dstrukt I am thinking similar experience we're considering for the app, users should be able to add any of the compatible strings and we should be smart on the backend and recognize what it is and act accordingly.
